src/modules/fw_pos_control_l1/fw_pos_control_l1_main.cpp

```diff
@@ -2338,14 +2338,20 @@ FixedwingPositionControl::task_main()
 					_att_sp.pitch_body = math::constrain(_att_sp.pitch_body, -_parameters.man_pitch_max_rad, _parameters.man_pitch_max_rad);
 				}
 
-				/* lazily publish the setpoint only once available */
-				if (_attitude_sp_pub != nullptr) {
-					/* publish the attitude setpoint */
-					orb_publish(_attitude_setpoint_id, _attitude_sp_pub, &_att_sp);
-
-				} else if (_attitude_setpoint_id) {
-					/* advertise and publish */
-					_attitude_sp_pub = orb_advertise(_attitude_setpoint_id, &_att_sp);
+				if (!(_control_mode.flag_control_offboard_enabled &&
+				      !(_control_mode.flag_control_position_enabled ||
```

Comment:
I think this will work for altitude control. tecs will provide a pitch setpoint, so the L1 controller should publish. The climb rate control flag will prevent the attitude controller from overwriting from the sticks. I'm not sure where the roll command comes from here, but I don't think this update changes that.

Am I missing something there? I'm away from my HiL setup right now so I can't test it.